server/api/balancer.go

```diff
@@ -61,6 +63,75 @@ func (h *balancerHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	h.rd.JSON(w, http.StatusOK, balancersInfo)
 }
 
+type operator struct {
+	Name     string `json:"name"`
+	RegionID uint64 `json:"region_id"`
+	StoreID  uint64 `json:"store_id"`
+	PeerID   uint64 `json:"peer_id"`
+}
+
+func newOperator(cluster *server.RaftCluster, m json.RawMessage) (uint64, server.Operator, error) {
+	op := &operator{}
+	if err := json.Unmarshal(m, op); err != nil {
+		return 0, nil, errors.Trace(err)
+	}
+
+	var (
+		operator server.Operator
+		err      error
+	)
+	switch op.Name {
+	case "add_peer":
+		operator, err = cluster.NewAddPeerOperator(op.RegionID, op.StoreID)
+	case "remove_peer":
+		operator, err = cluster.NewRemovePeerOperator(op.RegionID, op.PeerID)
+	default:
+		return 0, nil, errors.Errorf("invalid operator %v", op)
+	}
+	if err != nil {
+		return 0, nil, errors.Trace(err)
+	}
+
+	return op.RegionID, operator, nil
+}
+
+func (h *balancerHandler) Post(w http.ResponseWriter, r *http.Request) {
+	cluster, err := h.svr.GetRaftCluster()
+	if err != nil {
+		h.rd.JSON(w, http.StatusInternalServerError, err)
+		return
+	}
+	if cluster == nil {
+		h.rd.JSON(w, http.StatusOK, nil)
```

Comment:
I think I should open another pull request to deal with the `GetRaftCluster` and errors first.